hydra_base/lib/scenario.py

```diff
@@ -525,52 +535,96 @@ def _get_dataset_as_dict(rs, user_id):
     if rs.dataset is None:
         return None
 
-    dataset = deepcopy(rs.dataset.__dict__)
+    dataset = JSONDataset(rs.dataset)
 
     dataset['metadata'] = {}
 
-    del dataset['_sa_instance_state']
-
     try:
         rs.dataset.check_read_permission(user_id)
     except PermissionError:
            dataset['value']      = None
            dataset['metadata'] = {}
 
-    for m in rs.dataset.metadata:
-        dataset['metadata'][m.key] = m.value
+    #for m in rs.dataset.metadata:
+    #    dataset['metadata'][m.key] = m.value
 
     return dataset
 
-def _get_as_obj(obj_dict, name):
+def _get_scenario_network_resources(scenario_id, scenario_i=None):
     """
-        Turn a dictionary into a named tuple so it can be
-        passed into the constructor of a complex model generator.
+        Inspect the network of the given scenario and return a mapping of ID to Name
+        for each element in that network
```

Comment:
```suggestion
        for each element in that network, categorised by resource type.
```

Because the returned `mapping_dict` contains e.g. `mapping_dict['NODE'][id]` rather than ids directly.